Also install moose_test for apptainer moose, get installed framework tests working

loganharbour opened the pull request:
Closes #26463

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1975111/) on 96e1ddf wanted to post the following:

The following commit(s) contain fixups:

- 96e1ddf2f7: fixup! d61b61210bc2584b7c45004aaff714fd3e5000cf

loganharbour committed:
Enable loading of dynamic installable dirs

loganharbour committed:
Rework get_repo_revision to use env vars

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1975244/) on b1d54c0 wanted to post the following:

The following commit(s) contain fixups:

- 96e1ddf2f7: fixup! d61b61210bc2584b7c45004aaff714fd3e5000cf

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1975314/) on d864e36 wanted to post the following:

The following commit(s) contain fixups:

- d864e36e7a: fixup! 7fc4402888cfbca86ee08337aa88032fc66b0dae
- 96e1ddf2f7: fixup! d61b61210bc2584b7c45004aaff714fd3e5000cf

loganharbour committed:
Only split them if there are actually directories

loganharbour committed:
Capture variables so that we get the right ones

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1975629/) on 0809396 wanted to post the following:

The following commit(s) contain fixups:

- d864e36e7a: fixup! 7fc4402888cfbca86ee08337aa88032fc66b0dae
- 96e1ddf2f7: fixup! d61b61210bc2584b7c45004aaff714fd3e5000cf

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1975806/) on 6b68495 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/26464/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1976876/) on 7143c84 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>3833a5</th>
      <th colspan=4>#26464 7143c8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.07%</td>
      <td>85.07%</td>
      <td>+0.00%</td>
      <td>90.91%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>98390</td>
      <td>98411</td>
      <td>+21</td>
      <td>30</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>17264</td>
      <td>17266</td>
      <td>+2</td>
      <td>3</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26464/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26464/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/26464/coverage/xfem)

</details>



This comment will be updated on new commits.

loganharbour committed:
Skip hit cli tests when out of tree

refs #26479

loganharbour committed:
Disable out-of-tree tests that use find_moose_executable

refs #26480

loganharbour committed:
Remove out-of-tests mesh dependencies

This is so that the out-of-tree test copy will work, and therefore
these will work in an out-of-tree install

loganharbour committed:
Catch non-zero returns and also build moose_test for moose

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1978922/) on a614b7c wanted to post the following:

The following commit(s) contain fixups:

- 146112e891: fixup! 8d40788c89c31e3ff871a19ddf67ed7d26396ea7

loganharbour committed:
Cleanup run_tests function to catch any non-zero returns

loganharbour committed:
Enable verbosity in test run

loganharbour committed:
Actually run the binary

permcody commented on framework/include/base/MooseApp.h:
```diff
@@ -109,14 +109,13 @@ class MooseApp : public ConsoleStreamInterface,
+   * If \p real, then return the "real" binary name in the event that
+   * the binary is a symlink
+   */
+  std::string appBinaryName(const bool real = false) const;
```
You dropped virtual. Are we sure we never need to override this? What about SAM? Do we have any special cases?

permcody replied:
I will approve after talking with you and verifying that this is intentional.

loganharbour committed:
Finally fix the return flag logic

loganharbour committed:
Copy inputs into a folder based on the command line executable name

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1980059/) on 628c3c4 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/26464/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/26464/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 9157a263b5664e8d7d81a2f3eeb9fdc69e3629b0`

loganharbour commented:
@permcody this is ready again.

Difference from the last review is a force push that removed the first commit, replaced it with https://github.com/idaholab/moose/pull/26464/commits/ff61406e4c90928a9cda6e52e805d7de11c31cb2. I did a poor job with the original handling of copying inputs into a directory that matches the command line binary name (that is, when we do moose-opt -> combined-opt, put tests into a `moose` directory).
